models/path.py

```diff
@@ -149,6 +150,65 @@ def get_best_paths(cls, circuit):
         for path in cls.get_paths(circuit):
             yield cls.create_path(path["hops"])
 
+    @classmethod
+    def get_disjoint_paths(cls, circuit, unwanted_path):
+        """Computes the maximum disjoint paths from the unwanted_path for a EVC
+
+        Maximum disjoint paths from the unwanted_path are the paths from the
+        source node to the target node that share the minimum number os links
+        contained in unwanted_path. In other words, unwanted_path is the path
+        we want to avoid: we want the maximum possible disjoint path from it.
+        The disjointness of a path in regards to unwanted_path is calculated
+        by the complementary percentage of shared links between them. As an
+        example, if the unwanted_path has 3 links, a given path P1 has 1 link
+        shared with unwanted_path, and a given path P2 has 2 links shared with
+        unwanted_path, then the disjointness of P1 is 0.67 and the disjointness
+        of P2 is 0.33. In this example, P1 is preferable over P2 because it
+        offers a better disjoint path. When two paths have the same
+        disjointness they are ordered by 'cost' attributed as returned from
+        Pathfinder. When the disjointness of a path is equal to 0 (i.e., it
+        shares all the links with unwanted_path), that particular path is not
+        considered a candidate.
+
+        Parameters:
+        -----------
+
+        circuit : EVC
+            The EVC providing source node (uni_a) and target node (uni_z)
+
+        unwanted_path : Path
+            The Path which we want to avoid.
+
+        Returns:
+        --------
+        paths : generator
+            Generator of unwanted_path disjoint paths. If unwanted_path is
+            not provided or empty, we return an empty list.
+        """
+        unwanted_links = [
+            (link.endpoint_a.id, link.endpoint_b.id) for link in unwanted_path
+        ]
+        if not unwanted_links:
+            return None
+
+        paths = cls.get_paths(circuit, max_paths=settings.DISJOINT_PATH_CUTOFF)
```

Comment:
Minor suggestion for testability, if you want:

When injecting external dependencies like from settings, if you pass it as an argument, for example:

```
def get_disjoint_paths(cls, circuit, unwanted_path, max_paths=settings.DISJOINT_PATH_CUTOFF)
```

then it enhances testability without having to mock `settings.*`, (or alternatively to initialize something similar to `cls.max_get_paths_paths = settings.DISJOINT_PATH_CUTOFF`, but in this case it probably won't help that much). That would allow at unit test level to ensure that `get_paths` is being called with the expected arg that will be passed to pathfinder and then we don't need to test that it's actually cutting off since it's pathfinder's responsibility, but as a client mef_eline ensures that it's passing it as expected and asserting `assert_called_with`

Reply:
Great suggestion! thanks a lot, @viniarck . I've addressed your suggestion in commit 455d904.